haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Tuple, Dict
+
+from dateutil.parser import parse, ParserError
+
+from haystack.nodes.ranker.base import BaseRanker
+from haystack.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class RecentnessRanker(BaseRanker):
+    outgoing_edges = 1
+
+    def __init__(
+        self,
+        date_identifier: str,
+        weight: float = 0.5,
+        top_k: Optional[int] = None,
+        method: str = "reciprocal_rank_fusion",
```

Comment:
To make this more clear to users we can use something called `Literal` to specify the accepted options for `method` in the `__init__` signature. For example,
```python
# Add this to the top of the file
from typing import Literal

# Replace the method variable definition
def __init__(
    self,
    date_identifier: str, 
    weight: float = 0.5,
    top_k: Optional[int] = None,
    method: Literal["reciprocal_rank_fusion", "score"] = "reciprocal_rank_fusion",
):
```
This tells us directly in the function signature the accepted options without needing to read the doc strings.